src/picture.rs

```diff
@@ -294,14 +294,17 @@ pub(crate) unsafe fn rav1d_thread_picture_alloc(
     if res.is_err() {
         return res;
     }
-    let _ = mem::take(&mut (*c).itut_t35);
+    let _ = mem::take(&mut *(*c).itut_t35.write().unwrap());
```

Comment:
If this is the only place we write to `itut_t35`, dropping it, we may be able to avoid the `RwLock` once ownership is better.  Can you add a comment on the `itut_t35` field for that?

Reply:
I looked at that, and I think we need an immutable context at this point. Not certain, I'll put a comment.

Reply:
`fn rav1d_thread_picture_alloc` calls `fn picture_alloc_with_edges`, which calls `alloc_picture_callback`, which as #677 notes, is always called on the main thread.  Maybe all of `Rav1dContext` should be in a `Mutex`/`RwLock` here?

Reply:
We don't want to take a lock around the entire structure, as we might be doing something else with the context on a different thread, no? Also, we write to itut_t35 in several places (moving it from the context to the frame).

Reply:
I've removed the lock for itut_t35 based on your suggestions elsewhere.